### ISSUE TITLE
Suggestion: Automatic Database creation for Postgres

### ISSUE THREAD

**Issue by Immortalin** [strapi/strapi#3226]
**Informations**
- **Node.js version**: 10 
- **NPM version**: 6.9.0
- **Strapi version**: 3-alpha-26.2
- **Database**:  Postgres
- **Operating system**: Ubuntu Linux
- **(Optional) Link to your Project**: N/A

**What is the current behavior?**
Selecting postgres during setup stage requires inputting a database name. It is not sufficient for a db to be running and connected. User has to use `psql` to manually create DB. Ref: https://github.com/strapi/strapi/issues/355

**Steps to reproduce the problem**
Create a project, choose postgres as database.

**What is the expected behavior?**
Test postgres loigin, on success continue with projection creation. db generation is automatic and transparent to user, perhaps with a line printed out stating that a database was created.

**Suggested solutions**
As long as postgres login works, the required db should be automatically created if one was not specified, perhaps with a name generated from the project name.

**Comment by derrickmehaffy** [comment]
@Immortalin this is normal and most likely will not change. Providing admin/root level account information in order to create a sub account in my opinion is a security risk. The same applies to both MySQL and Mongo (with authentication enabled, as it should be).

Gone are the days of giving your database root/admin account to a php application to have it create an account for you.

I'll let @lauriejim decided if this should be closed, but I very strongly vote no to this. Also it is more of a feature request than a bug.

**Comment by Immortalin** [comment]
For Rails and Django etc. this never seemed to be an issue. Especially for local dev envs

**Comment by Immortalin** [comment]
You can ask user to create account manually only if automated account creation fails. There is no risk when the PostgreSQL is running in a docker container.

**Comment by derrickmehaffy** [comment]
> For Rails and Django etc. this never seemed to be an issue. Especially for local dev envs

Anytime you are passing root/admin credentials is not a good thing, as the saying goes if your friend jumps off a bridge do you follow?

Locally sure, very little risk. But the goal of Strapi is to not run locally, and configuring a database along with the proper user and permissions is not difficult and certainly helps users understand what they are using. Typically all SQL based databases function in a similar way:

1) Create the database
2) Create the User
3) Grant Permissions to the user

```
Postgres:
sudo -u postgres psql
postgres=# create database mydb;
postgres=# create user myuser with encrypted password 'mypass';
postgres=# grant all privileges on database mydb to myuser;

MySQL: 
mysql -u someadmin -p
create database mydb;
create user@localhost identified by 'somepassword';
grant all privileges on mydb.* to user@localhost;
```

I see very little benefit to adding a step that creates the database considering Strapi should not create the user as doing so would require admin level permissions, and in the case of cloud databases which are becoming a lot more popular (quite a few users are using them) it makes very little sense to build this in and has the chance to create a security issue.

Any company or large project using Strapi would have a systems admin/engineer and any small group/single user could benefit from learning a bit devops.

**Comment by derrickmehaffy** [comment]
And no:

> Considering how Docker is almost always used for deploying strapi

That is not the case, and when it comes to databases I personally **HIGHLY** suggest that you do **NOT** dockerize your database. But my bias aside the bulk of the users I see and help in Slack and here on github are not using docker with Strapi.

**Comment by Immortalin** [comment]
What's wrong with dockerizing database? The data is stored in a volume anyways.

**Comment by derrickmehaffy** [comment]
https://myopsblog.wordpress.com/2017/02/06/why-databases-is-not-for-containers/
https://vsupalov.com/database-in-docker/
https://thehftguy.com/2016/11/01/docker-in-production-an-history-of-failure/

TLDR: You shouldn't put things in docker just to put things in docker.

Docker adds a whole other level of network layer you have to deal with, which makes scaling, monitoring, caching, and just general performance a massive pain in the ass.

But this is getting off topic.

**Comment by Immortalin** [comment]
Would you feel the same way about BSD Jails?

**Comment by derrickmehaffy** [comment]
Jails are far more secure but we are getting off topic.

Automatically creating a database provides very little benefit as to do so the DB user would need to have DB admin level permissions.

I'm marking this as closed.
